src/System.Diagnostics.Tracing/src/System/Diagnostics/Tracing/EventCounter.cs

```diff
@@ -92,10 +108,8 @@ private void Enqueue(float value)
                 {
                     // It is possible that two threads both think the buffer is full, but only one get to actually flush it, the other
                     // will eventually enter this code path and potentially calling Flushing on a buffer that is not full, and that's okay too.
-                    lock (_bufferedValues)
-                    {
+                    lock (this) // Lock the counter
```

Comment:
We can lock on something else (like _bufferdValues), and I will do that if you feel strongly about it.   But I feel that desire not to lock on 'this' stemmed from code access security days, when we were trying to prevent issues from active hostile clients.   I don't think we live in that world anymore, and using 'this' is something we should not be fighting anymore.

But I will change it if you want me to.